client/me/purchases/purchases-site/index.tsx

```diff
@@ -18,32 +17,40 @@ import type { StoredCard } from 'calypso/my-sites/checkout/composite-checkout/ty
 
 import './style.scss';
 
-export default function PurchasesSite( {
-	getManagePurchaseUrlFor = managePurchase,
-	isPlaceholder,
-	siteId,
-	purchases,
-	name,
-	slug,
-	cards,
-	showSite = false,
-}: {
-	getManagePurchaseUrlFor: ( slug: string, purchaseId: number ) => string;
-	isPlaceholder: boolean;
-	siteId: number;
-	purchases: Purchase[];
-	name: string;
-	slug: string;
-	cards: StoredCard[];
-	showSite: boolean;
-} ): JSX.Element {
-	const site = useSelector( ( state ) => getSite( state, siteId ) );
-	if ( isPlaceholder ) {
+export default function PurchasesSite(
+	props:
+		| {
+				isPlaceholder: true;
+				siteId?: number;
+		  }
+		| {
+				getManagePurchaseUrlFor: ( slug: string, purchaseId: number ) => string;
+				isPlaceholder?: false;
+				siteId: number;
+				purchases: Purchase[];
+				name: string;
+				slug: string;
+				cards: StoredCard[];
+				showSite?: boolean;
```

Comment:
this is totally fine, and helps move the project forward. it's also a good opportunity to create names for types. sometimes the anonymous type unions can be confusing and a name could clear things up.

```ts
interface PlaceholderProps {
	isPlaceholder: true;
	siteId?: number;
}

interface LoadedProps {
	…
}

type Props = PlaceholderProps | LoadedProps;

export default function PurchaseSite(props: Props) { … }
```

Reply:
> the anonymous type unions can be confusing and a name could clear things up

I've realized that a lot lately. It also helps make error message more clear.

Reply:
That's interesting. Where is the other `Props` coming in? Is it from a parent component? I've argued the case for avoiding descriptive React component prop names for making things noisy 😆

Reply:
I think I've mainly seen it WRT generics in things like HOCs which rely on merging the props of two components.

Reply:
that makes sense. would be nice if those HOCs didn't use `Props` and `State` for this very reason